### ISSUE TITLE
Routing between default layout and persistent layout doesn't slot in the page

### ISSUE THREAD

**Issue by kobe-ra** [inertiajs/inertia#1733]
### Version:

- "@inertiajs/svelte": "^1.0.14",

### Describe the problem:

I have MasterLayout that is for logged in users. Then there is GuestLayout for guests. When user logs out and is redirected to Login page, GuestLayout loads, but the Login page doesn't. I then manually refresh the browser window, everything loads normally on the Login page. After submitting login form with useForm helper, user is redirected to Home page, but again - MasterLayout is loaded and works properly, while Page itself is not present at all.
[Video](https://youtu.be/EvUrAtNy3jo)

### Steps to reproduce:

app.js
```
import "./bootstrap";
import MasterLayout from "./Layouts/MasterLayout.svelte";

import { createInertiaApp } from "@inertiajs/svelte";

createInertiaApp({
    resolve: (name) => {
        const pages = import.meta.glob("./Pages/**/*.svelte", { eager: true });
        let page = pages[`./Pages/${name}.svelte`];
        console.log(page)
        // return { ...page, layout: typeof page.layout === 'undefined' ? MasterLayout : page.layout }
        return {
            default: page.default,
            // layout: (_) => page.layout || MasterLayout,
            layout: page.layout || MasterLayout,
        };
    },
    setup({ el, App, props }) {
        // new App({ target: el });
        new App({ target: el, props });
    },
});
```

GuestLayout.svelte
```
<script>
    console.log('guest layout')
</script>

<slot />
```

Login.svelte
```
<script context="module">
    export { default as layout } from "@/Layouts/GuestLayout.svelte";
</script>

<script>
    import { useForm } from "@inertiajs/svelte";
    import Input from "@/Components/Forms/Input.svelte";

    let form = useForm({
        login: "",
        password: "",
    });

    const submit = () => {
        $form.post(route("login"));
    };

    console.log("login page");
</script>

<div class="container h-screen">
    ...
</div>
```

MasterLayout.svelte
```
<script>
    import Navigation from "@/Layouts/Navigation.svelte";

    console.log('master layout')
</script>

<div id="app">
    <Navigation>
        <slot />
    </Navigation>
</div>
```

Home.svelte
```
<script>
    console.log('home page')
</script>

<svelte:head>
    <title>{import.meta.env.VITE_APP_NAME}</title>
</svelte:head>

data data
```

**Comment by Josefikos** [comment]
i'm facing same issue for a while

**Comment by Josefikos** [comment]
related to this: https://github.com/inertiajs/inertia/issues/1621

Forcing svelte to the version of 4.0.0 solved problem

**Comment by driesvints** [comment]
Seems the following PR will solve this so let's see how it goes: https://github.com/inertiajs/inertia/pull/1763. Thanks all.

**Comment by reinink** [comment]
This has been fixed in #1763 has been released as part of [v1.0.16](https://github.com/inertiajs/inertia/releases/tag/v1.0.16) 👍
